### ISSUE TITLE
CA2000 False Positive (regression) when method returns IDisposable result returned by another method

### ISSUE THREAD

**Issue by RobSiklos** [dotnet/roslyn-analyzers#2583]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.3

#### Diagnostic ID

[CA2000: Dispose objects before losing scope](https://docs.microsoft.com/en-ca/visualstudio/code-quality/ca2000-dispose-objects-before-losing-scope)

#### Repro steps

1. Analyze the code below:

```csharp
public class IssueWithCA2000
{
    [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1034:Nested types should not be visible")]

    public sealed class MyResult : IDisposable
    {
        public void Dispose() { }
    }

    public MyResult DoSomething()
    {
        try { }
        catch (ArgumentException)
        {
            // CA2000 reported here
            // Call System.IDisposable.Dispose on object created by 'this.CreateResponse()' before all references to it are out of scope
            return this.CreateResponse();
        }

        return null;
    }

    private MyResult CreateResponse() { return new MyResult(); }
}
```

#### Expected behavior
No warning, because the method is returning the IDisposable object.

#### Actual behavior
CA2000 violation.

**Comment by RobSiklos** [comment]
Verified this issue no longer appears with https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.4-beta1.19318.2+bc3ca30e.

Thanks!
